Marlin/src/inc/SanityCheck.h

```diff
@@ -930,6 +930,17 @@ static_assert(X_MAX_LENGTH >= X_BED_SIZE && Y_MAX_LENGTH >= Y_BED_SIZE,
   #error "MESH_EDIT_GFX_OVERLAY requires AUTO_BED_LEVELING_UBL and a Graphical LCD."
 #endif
 
+#if ENABLED(G29_RETRY_AND_RECOVER)
+  #if !HAS_LEVELING
+    #error "G29_RETRY_AND_RECOVER requires automatic bed leveling"
+  #elif !ENABLED(OLDSCHOOL_ABL)
+    #error "G29_RETRY_AND_RECOVER currently only supports ABL"
+  #endif
+  #ifndef G29_MAX_RETRIES
+    #define G29_MAX_RETRIES 0
```

Comment:
This define should go into the code file that uses it, since only one file does.